Sources/TuistSupport/Logging/DetailedLogHandler.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+import Logging
+
+public struct DetailedLogHandler: LogHandler {
+    public let label: String
+
+    private var stdout: LogHandler
+    private var stderr: LogHandler
+
+    public init(label: String) {
+        self.init(label: label, logLevel: .info)
+    }
+
+    public init(label: String, logLevel: Logger.Level) {
+        self.label = label
+        stdout = StreamLogHandler.standardOutput(label: label)
+        stdout.logLevel = logLevel
+        stderr = StreamLogHandler.standardOutput(label: label)
+        stderr.logLevel = logLevel
+    }
+
+    public func log(
+        level: Logger.Level,
+        message: Logger.Message,
+        metadata: Logger.Metadata?,
+        file: String, function: String, line: UInt
+    ) {
+        var log: String = "\(timestamp()) \(level.rawValue) \(label)"
+
+        let mergedMetadata = metadata.map { self.metadata.merging($0, uniquingKeysWith: { $1 }) } ?? self.metadata
+
+        if mergedMetadata.isEmpty == false {
+            log.append(mergedMetadata.pretty)
```

Comment:
Not sure I follow the logic here :( - in case there is no metadata add the `pretty` metadata item?

Reply:
Ill write an extension `isNotEmpty` so it reads nicer

Reply:
```swift
extension Collection {
    var isNotEmpty: Bool { !isEmpty }
}
```

so it reads like:
```swift
if mergedMetadata.isNotEmpty {
    log.append(mergedMetadata.prettyDescription)
}
```

Reply:
thanks - the rename helped I initially confused `.pretty` here with the other `.pretty` metadata item used for `logger.pretty`